blocktx/store/sql/get_block_gaps.go

```diff
@@ -17,7 +17,9 @@ func (s *SQL) GetBlockGaps(ctx context.Context, blockHeightRange int) ([]*store.
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	q := `SELECT missing_blocks.missing_block_height, prevhash AS hash FROM blocks
+	q := `
```

Comment:
I still don't understand why cannot we tie blocks to the previous one? Let's not process block we don't already have processed previous block so it guarantees that we don't have gaps? Leaving gaps is very dangerous and uncomfortable for users. Imagine not giving information that something is already spent.

As a user I would prefer the most up to date state to the state that might be missing deposits or withdrawals.

Reply:
The gap filling mechanism in place has proved working well. There are no gaps anymore. The mechanism in place which follows requests all previous blocks was not good enough: https://github.com/bitcoin-sv/arc/blob/56b20194990d79c5cf38cd8d42b94953b0cd4cca/blocktx/peer_handler.go#L587
because if for example `blocktx` is down for a while and missed some blocks it can well happen that there are gaps.

The changed query now includes not only the gaps, but also those blocks without a `processed_at` date. Blocks without this date mean, that the block processing was started, but never finished. These blocks should also be considered as gaps. This can also always happen, when for example `blocktx` gets a new deployment while it is processing a block and can't finish the processing.